### PR TITLE
Add GitHub Actions workflow for static code analysis

### PR DESCRIPTION
**Pull request by antoineco**
Integrates with GitHub Actions to provide immediate feedback on PRs when issues are found via static code analysis.

The only "static analysis" performed right now is linting (which includes gofmt checks). I can imagine us adding more in the near future, e.g. checks for EOF without newline character at the end of Kubernetes manifests, Dockerfiles, etc.

I deliberately pushed a broken commit, which I will revert upon approval, to demonstrate what is reported when issues are found.

### CODE REVIEW

File: pkg/routing/adapter/splitter/adapter.go [hunk 1]
```diff
@@ -165,10 +165,7 @@ func (h *Handler) split(path string, e *event.Event) []*event.Event {
 	}
 	for _, v := range val.Array() {
 		newCE := cloudevents.NewEvent()
-		if err := newCE.SetData(cloudevents.ApplicationJSON, v.Raw); err != nil {
-			h.logger.Errorw("Failed to set event data", zap.Error(err))
-			continue
-		}
+		newCE.SetData(cloudevents.ApplicationJSON, v.Raw)
```
Comment (sebgoa) [inline pkg/routing/adapter/splitter/adapter.go:168]: just to be picky because you are sometimes :) why did you fix this in this PR about github actions ?

Comment (antoineco) [reply pkg/routing/adapter/splitter/adapter.go:168]: It's explained in the description.

Comment (sebgoa) [reply pkg/routing/adapter/splitter/adapter.go:168]: ok intentional fmt breaks

**Review by odacremolbap — APPROVED** [review]

**Comment by antoineco** [comment]
Because I removed the broken PR, here is a screenshot of the (previous) automated code review:

image

**Pull request merged**